esphome/components/http_request/http_request_arduino.cpp

```diff
@@ -121,6 +129,18 @@ std::shared_ptr<HttpContainer> HttpRequestArduino::start(std::string url, std::s
     // Still return the container, so it can be used to get the status code and error message
   }
 
+  container->response_headers_ = {};
+  for (int i = 0; i < container->client_.headers(); i++) {
+    const auto *header_name = container->client_.headerName(i).c_str();
```

Comment:
does it really need to be a c_str()? Or is it because the function returns an "arduino" `String`?
```suggestion
    const auto header_name = container->client_.headerName(i);
```